### PR TITLE
Fixed issue with cat in fx backend

### PR DESCRIPTION
**Pull request by suryasidd**
Fixed issue with cat in fx backend that was causing perf degradation

Added new op batch_norm_legit_no_training which was introduced in Pytorch 2.1.0 which is just batch_norm op with no training argument

### Tickets:
 - https://jira.devtools.intel.com/browse/CVS-124095

### CODE REVIEW

**Comment by suryasidd** [comment]
build_jenkins

File: src/frontends/pytorch/src/op/batch_norm.cpp [hunk 1]
```diff
@@ -63,7 +67,13 @@ OutputVector translate_batch_norm(const NodeContext& context) {
         bias = broadcast_const_to_channel_dim(context, input, zero_f);
     }
     // index 3 running_mean and index 4 running_var can be none for training case only, check that not training before
-    auto training = context.const_input<bool>(5);
+    bool training;
+    if(is_bn_no_training){
+        training = false;
+    }
+    else{
+        training = context.const_input<bool>(5);
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/batch_norm.cpp:75]: In my opinion better would be to extract training argument before calling `translate_batch_norm_common`

**Review by mvafin — CHANGES_REQUESTED** [review]
Please fix code style

File: src/frontends/pytorch/src/op/batch_norm.cpp (hunk 1 above)
Comment (suryasidd) [reply src/frontends/pytorch/src/op/batch_norm.cpp]: Makes sense Maxim, then i can check for the num of inputs and get the value for epsilon. Made the changes

**Comment by suryasidd** [comment]
build_jenkins

File: src/frontends/pytorch/src/op/batch_norm.cpp [hunk 2]
```diff
@@ -92,14 +95,31 @@ OutputVector translate_batch_norm(const NodeContext& context) {
...
+    }
     // Input with index 8 is flag "cudnn_enabled" we can ignore it
     return {context.mark_node(
         std::make_shared<v5::BatchNormInference>(input, weight, bias, running_mean, running_var, epsilon))};
 };
 
-OutputVector translate_batch_norm_fx(const NodeContext& context) {
-    auto output = translate_batch_norm(context);
+OutputVector translate_batch_norm(const NodeContext& context) {
+    auto training = context.const_input<bool>(5);
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/batch_norm.cpp:111]: you need to check num inputs that input 5 exist

**Review by mvafin — COMMENTED** [review]
Please check num inputs in individual translator, not in common code

**Review by mvafin — APPROVED** [review]

**Comment by mvafin** [comment]
build_jenkins

**Comment by suryasidd** [comment]
build_jenkins

**Comment by suryasidd** [comment]
build_jenkins

**Pull request merged**